static/app/components/events/interfaces/spans/spanTree.tsx

```diff
@@ -739,6 +799,114 @@ class SpanTree extends Component<PropType> {
   }
 }
 
+type SpanRowProps = ListRowProps & {
+  addSpanRowToState: (
+    spanId: string,
+    spanRow: React.RefObject<HTMLDivElement>,
+    treeDepth: number
+  ) => void;
+  cache: CellMeasurerCache;
+  removeSpanRowFromState: (spanId: string) => void;
+  spanContextProps: SpanContext.SpanContextProps;
+  spanTree: SpanTreeNode[];
+};
+
+function SpanRow(props: SpanRowProps) {
+  const {
+    index,
+    parent,
+    style,
+    columnIndex,
+    spanTree,
+    cache,
+    spanContextProps,
+    addSpanRowToState,
+    removeSpanRowFromState,
+  } = props;
+
+  const rowRef = useRef<HTMLDivElement>(null);
+  const spanNode = spanTree[index];
+
+  // Lifecycle management for row refs, we need to separately do this in useLayoutEffect since
+  // we won't have access to the refs in useEffect
```

Comment:
Ah right sorry I didn't make it very clear, it's specifically because we don't have access to the refs when we are unmounting. From [React's useLayoutEffect docs](https://reactjs.org/docs/hooks-reference.html#uselayouteffect):

*"Updates scheduled inside useLayoutEffect will be flushed synchronously, before the browser has a chance to paint."*

In `useEffect`, the return function for cleanup wasn't able to remove the ref from the map since the component no longer had access to it, since the return function is executed after the browser paints and so the DOM node is removed.

I can add this to the comment to make it more clear